### ISSUE TITLE
Audit ARIA in HTML global aria-* allowances for null-role elements (e.g. form) in aria-prohibited-attr

### ISSUE THREAD

**Issue by chutchins25** [dequelabs/axe-core#5244]
Follow-up to #5185, which fixed `aria-prohibited-attr` wrongly flagging `aria-label`/`aria-labelledby` on an unnamed `<section>`/`<aside>` (elements that resolve to a `null` role but still allow global `aria-*` attributes per ARIA in HTML).

The same pattern affects other elements whose implicit role is name-dependent and resolves to `null` when unnamed. For example, `<form>`:

```js
// lib/commons/standards/implicit-html-roles.js
form: vNode => (hasAccessibleName(vNode) ? 'form' : null)
```

So `<form aria-labelledby="…">` with an empty/unresolved reference has the same false positive today.

## Ask

Audit the ARIA in HTML "Global `aria-*` attributes" allowances against axe's `aria-prohibited-attr` handling, and add any other elements that:

1. resolve to a `null` role when they have no accessible name, and
2. are permitted global `aria-*` attributes by the spec,

to the `elementsAllowedAriaLabel` option (or a more general mechanism if the list grows large).

Candidates to confirm against the spec: `form`, and any other name-dependent-implicit-role elements. (`header`/`footer` map to `null` only when scoped, but their allowance is name-independent — confirm during the audit.)

Related: #5185

**Comment by chutchins25** [comment]
Resolved by #5259, which reworked `aria-prohibited-attr` to determine naming from the html-elm spec when an element resolves to its implicit role. An unnamed `form` resolves to its implicit role and isn't `namingProhibited`, so it is allowed to be named under the new mechanism — the broader global-attr allowance this issue asked for is covered. Closing.
